### ISSUE TITLE
A3 Thermal Improvement break the drone-noise effect for camera view

### ISSUE THREAD

**Issue by Crowdedlight** [Crowdedlight/Crows-Electronic-Warfare#78]
Reported that when using the A3 Thermal Improvement and hitting the key to go into their "camera view" of thermal, the noise effect is removed and doesn't come back. 

Not sure what is causing this. I guess the A3 Thermal Improvement resets effects to do their own handling, so not sure if we can cooperative with the mod for not breaking the drone noise effect...

**Comment by Crowdedlight** [comment]
@b-mayr-1984 just FYI, in case you let people use the A3 Thermal Improvement mod for the scenario/event.

**Comment by b-mayr-1984** [comment]
### A3 Thermal Improvement
> Reported that when using the A3 Thermal Improvement and hitting the key to go into their "camera view" of thermal, the noise effect is removed and doesn't come back.
> 
> Not sure what is causing this. I guess the A3 Thermal Improvement resets effects to do their own handling, so not sure if we can cooperative with the mod for not breaking the drone noise effect...

Might be related to `GVAR(FilmGrain_jamEffect);` only being created once and then manipulated constantly. Maybe this mod deletes all post processing effects as part of a reset routine etc.
I somewhat diverted from the classical implementation examples here where the effect variable is created directly in front of the actual manipulation. But I am just guessing here.

### Shoot and Scoot
> @b-mayr-1984 just FYI, in case you let people use the A3 Thermal Improvement mod for the scenario/event.

That mod is not part of the modest (and not part of a whitelist for optional mods. 
In addition the drones are all nerfed to have no thermal imaging at all (because this is not very TvT friendly). 
See https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/blob/cb4e672e49a3fb5cf4fe9ad6a834de4697021afb/init.sqf#L31

**Comment by Crowdedlight** [comment]
> In addition the drones are all nerfed to have no thermal imaging at all (because this is not very TvT friendly)
Ah ofc. Makes total sense. 

> Do we have RPT logs of the problem?
> ### Shoot and Scoot

Not currently. I will try to get the RPT from the mission from the drone operator. Have not tried to reproduce it, but he said it was fairly consistent when using the mod as the drone operator in that mission. When getting noise effect just hitting the shortcut twice. 

I will investigate further!

**Comment by Crowdedlight** [comment]
[Arma3_x64_2024-04-01_21-10-00.txt](https://github.com/Crowdedlight/Crows-Electronic-Warfare/files/14859581/Arma3_x64_2024-04-01_21-10-00.txt)

UAV operator believe this was the RPT for the incident. Haven't had time to check myself, but posting here for when I get around to it.

**Comment by b-mayr-1984** [comment]
> [Arma3_x64_2024-04-01_21-10-00.txt](https://github.com/Crowdedlight/Crows-Electronic-Warfare/files/14859581/Arma3_x64_2024-04-01_21-10-00.txt)
> 
> UAV operator believe this was the RPT for the incident. Haven't had time to check myself, but posting here for when I get around to it.

Might not be related, but some messages caught my attention:
```
21:58:55 [ACE] (nightvision) ERROR: no fog backed up
[..]
22:21:04 [CBA] (optics) INFO: Scripted camera restarted.
[..]
22:22:55 B Bravo 1-2:1 (UCAV Sentinel): Unknown action ActiveSensorsOn, target B Bravo 1-2:1
```

**Comment by Crowdedlight** [comment]
yeah, the scripted-camera restart I could imagine would reset all effects. Although not sure why it would not be applied again next iteration. 

I think might have to unpack the mod pbos and see how they do it. If I want to support that mod. For now I think I will add to readme that there is issues with that mod and it might break the player drone jamming. So people at least know.

**Comment by KapoDiGabinetto** [comment]
I can confirm no video feed degrade in thermal camera of the drone.Test with just this mod,cba and zeus enhanced.

**Comment by b-mayr-1984** [comment]
@KapoDiGabinetto what do you mean by that?
> Test with just this mod [..]

Do you mean `A3 Thermal Improvement`?
Or do you mean `Crows Electronic Warfare`?

**Comment by KapoDiGabinetto** [comment]
ops you are right ...without A3TI, just Crows Electronic Warfare;CBA;Ace;Zeus enhanced...sorry

**Comment by Crowdedlight** [comment]
Fair. I believe I tested it with night-vision, but might not have checked thermal and what happens if you swap between vision modes while the effect is in effect. 

I'll take a look at it when I get some more spare time.

**Comment by b-mayr-1984** [comment]
For some reason [Post Process Effects](https://community.bistudio.com/wiki/Post_Process_Effects) have no effect in TI view.
I tried with very large rendering priorities, but this made no difference.

Might be a BI bug.

A [SPOTREP from 2022](https://dev.arma3.com/post/spotrep-00105) is the only source I could find that mentions both thermal imaging and Post Process Effects. But they don't appear related in this post.

**Comment by Crowdedlight** [comment]
> For some reason [Post Process Effects](https://community.bistudio.com/wiki/Post_Process_Effects) have no effect in TI view. I tried with very large rendering priorities, but this made no difference.
> 
> Might be a BI bug.
> 
> A [SPOTREP from 2022](https://dev.arma3.com/post/spotrep-00105) is the only source I could find that mentions both thermal imaging and Post Process Effects. But they don't appear related in this post.

Interesting! 
Might need to ask BI/Ace Discord if they know anything about it.

**Comment by OverlordZorn** [comment]
Did you try to change the layer/priority of your applied effects?

Is it only with TI Improved mod or also with vanilla?

**Comment by b-mayr-1984** [comment]
@OverlordZorn 
> Did you try to change the layer/priority of your applied effects?

Yes, I tried  that (as stated).
> I tried with very large rendering priorities, but this made no difference.

### TI Improved mod
> Is it only with TI Improved mod or also with vanilla?

I only tested with Vanilla Arma. But original issue Report speaks of TI improvement Mod.

**Comment by b-mayr-1984** [comment]
> For some reason [Post Process Effects](https://community.bistudio.com/wiki/Post_Process_Effects) have no effect in TI view. I tried with very large rendering priorities, but this made no difference.
> 
> Might be a BI bug.
> 
> A [SPOTREP from 2022](https://dev.arma3.com/post/spotrep-00105) is the only source I could find that mentions both thermal imaging and Post Process Effects. But they don't appear related in this post.

@brettmayson reports he as the same issue with [AID mod](https://steamcommunity.com/sharedfiles/filedetails/?id=3228179079).
see https://discord.com/channels/976165959041679380/978362166119919626/1275140524176572549

**Comment by b-mayr-1984** [comment]
> Reported that when using the A3 Thermal Improvement and hitting the key to go into their "camera view" of thermal, the noise effect is removed and doesn't come back.
> 
> Not sure what is causing this. I guess the A3 Thermal Improvement resets effects to do their own handling, so not sure if we can cooperative with the mod for not breaking the drone noise effect...

@johnb432 reports that A3TI also causes issues with ACE3 functionality (e.g. in connection with `ace_common_oldIsCamera`)
see https://discord.com/channels/976165959041679380/978362166119919626/1275130058343977062

**Comment by johnb432** [comment]
For more info (text was written and posted by someone else on the A3TI discord):

>Hello. I tracked down a bug while using A3TI with ACE and Zeus. When the zeus character itself is unconscious, the fade to black screen stays in effect when in the zeus camera view. This is a problem since the zeus interface cannot be properly used during phased where the player wants to use zeus interface during that phase when they are incapacitated. I suspect the camera view EH in A3TI is colliding with the one from ACE. The respective functionality was fixed with an update of ACE during november, afair.

**Comment by b-mayr-1984** [comment]
> For some reason [Post Process Effects](https://community.bistudio.com/wiki/Post_Process_Effects) have no effect in TI view. I tried with very large rendering priorities, but this made no difference.
> 
> Might be a BI bug.

This bug is known/reported since 2014.
https://feedback.bistudio.com/T75717
